hibernate-core/src/main/java/org/hibernate/query/derived/AnonymousTupleEntityValuedModelPart.java

```diff
@@ -693,4 +693,9 @@ public boolean isSimpleJoinPredicate(Predicate predicate) {
 				? ( (TableGroupJoinProducer) delegate ).isSimpleJoinPredicate( predicate )
 				: false;
 	}
+
+	@Override
+	public EntityPersister asEntityPersister() {
+		return ( EntityPersister ) this;
```

Comment:
doesn't need a cast?

Reply:
Intellij IDEA warning me again.

Reply:
well IDEA was right, this cast can't possibly work. I've removed the method, and as expected all tests are still passing: that's because this isn't ever invoked, otherwise you'd have seen a classcast exception.

Same for all other IDEA warnings.